pandas/tests/arrays/string_/test_string.py

```diff
@@ -655,7 +663,11 @@ def test_isin(dtype, fixed_now_ts):
     tm.assert_series_equal(result, expected)
 
     result = s.isin(["a", pd.NA])
-    expected = pd.Series([True, False, True])
+    if dtype.storage == "python" and dtype.na_value is np.nan:
+        # TODO what do we want here?
```

Comment:
I think this is the best outcome

Reply:
Hmm OK - maybe we should just have this be a branch between `np.nan` and `pd.NA` then? 

> But given that with plain object dtype we currently also match pd.NA with None, I would maybe rather keep that behaviour?

I see the reasoning for it but I am hesitant to keep repeating this behavior. Seems like its really a quirk of how our historical Python object storage has worked, and I don't see that aging well as we move beyond it

Reply:
Thinking about this a bit more: when doing `ser.isin(["a", pd.NA])`, the user is providing a list here, i.e. not something with already a well defined dtype (numpy array or pandas array/series). 
So what I expect what would happen here is that the list of values is coerced to the calling `ser.dtype`. And then this should give the same result as `ser.isin(pd.array("a", pd.NA], ser.dtype))`.

`pd.array("a", pd.NA], ser.dtype)` will coerce the NA to a missing value for that dtype, so in case of the new string dtype, this is NaN.

(to be clear, what actually happens is clearly not matching what I describe above as what I would expect)

Reply:
I think I agree with @jorisvandenbossche description of the result (and object sucks...)